providers/oidc.go

```diff
@@ -23,6 +23,7 @@ type OIDCProvider struct {
 	Verifier             *oidc.IDTokenVerifier
 	AllowUnverifiedEmail bool
 	UserIDClaim          string
+	GroupClaim           string
```

Comment:
I think we potentially want to rename this -- I was confused at first, thinking this mirrored `UserIDClaim` that it gave an alternative claim in case a user wanted to use an alternative (ie UserIDClaim not defaulting to `email`).

But if I understand right, this is a group to search in the hardcoded rawClaims["groups"] for the presence of? And if missing, do rudimentary authorization logic based on the group membership?

In which case we probably want 2 flags, 1 denoting the claim if it isn't the standard `groups` -- another denoting groups that must be present in that claim.

Reply:
Ok this makes sense, hows about `--groups-claim` and `--user-group` however I am not sure if we should denote this only works with oidc by having something like `--oidc-groups-claim` and `--oidc-user-group`?

Reply:
Yeah -- `--groups-claim` makes sense for the claim name.  `--user-group` feels a little vague, maybe `required-group` (with a plural depending on implementation).

This potentially seems best as an array option where one of a list of groups needs to be present in the `groups` claim?

As far as OIDC, while you are only implementing this in that provider at the moment, I would hope we could potentially add this to a few other providers that might have this claim in their IDTokens someday (I only use OIDC myself, so I'm not positive what all the others IDTokens look like).

So leaving the flag as a broader name to match UserIDClaim makes sense.

Reply:
Sounds good to me! will get back to it this week and get those items wrapped up.